File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {

        int n = s.length();
        stack<char> stck;

        for(int i=0; i<n; i++)
        {
            if(s[i]=='(' || s[i]=='[' || s[i]=='{'){
                stck.push(s[i]);
            }
            else{
                if(stck.empty()){
                    return false;
                }
                else if(s[i]==')' && stck.top()!='(' || s[i]==']' && stck.top()!='[' ||
                        s[i]=='}' && stck.top()!='{')
                {
                    return false;
                }
                else{
                    stck.pop();
                }
            }
        }
        return stck.empty();        
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        for i in range(len(s)):
            if s[i] == '(' or s[i] == '[' or s[i] == '{':
                stack.append(s[i])
            elif s[i] == ')' and len(stack)!=0 and stack[-1] == '(' :
                stack.pop()
            elif s[i] == ']' and len(stack)!=0 and stack[-1] == '[' :
                stack.pop()
            elif s[i] == '}' and len(stack)!=0 and stack[-1] == '{':
                stack.pop() 
            elif (s[i] == ')' and len(stack)==0) or (s[i]==')' and len(stack)!=0 and stack[-1]!='('):
                return False
            elif s[i] == ']' and len(stack)==0 or (s[i]==']' and len(stack)!=0 and stack[-1]!='[') :
                return False
            elif s[i] == '}' and len(stack)==0 or (s[i]=='}' and len(stack)!=0 and stack[-1]!='{'):
                return False
        if len(stack)==0:
            return True
        else:
            return False